jdaviz/configs/imviz/wcs_utils.py

```diff
@@ -424,5 +456,38 @@ def _get_rotated_nddata_from_label(app, data_label, rotation_angle, refdata_shap
     data = app.data_collection[data_label]
     if data.coords is None:
         raise ValueError(f"{data_label} has no WCS for rotation.")
-    return _prepare_rotated_nddata(data.shape, data.coords, rotation_angle, refdata_shape,
-                                   wcs_only_key=app._wcs_only_label)
+
+    if cdelt_signs is None:
+        # check if East is to the left:
+        viewer = app.get_viewer('imviz-0')
+        wcs = viewer.state.reference_data.coords
+        if any(['lon' in name for name in wcs.world_axis_names]):
+            # works for gwcs:
+            [lon_axis] = [i for i, axis in enumerate(wcs.world_axis_names) if axis == 'lon']
+        else:
+            # works for FITS WCS:
+            [lon_axis] = [i for i, axis in enumerate(wcs.wcs.ctype) if axis.startswith('RA')]
+        offset_coords = [1 if lon_axis == i else 0 for i in range(2)]
+        origin_coords = [0, 0]
+        east_left = (
+            wcs.pixel_to_world_values(*offset_coords)[0] -
+            wcs.pixel_to_world_values(*origin_coords)[0]
+        ) < 0
+
+        if east_left:
+            cdelt_signs = None
+        else:
+            cdelt_signs = [-1 if lon_axis == i else 1 for i in range(2)]
+            if 'imviz-compass' in [item['name'] for item in app.state.tray_items]:
+                compass_plugin = app.get_tray_item_from_name('imviz-compass')

```

Comment:
don't love the deeper-level access here, but since this is acting directly on the app and not helper, agree this probably makes the most sense